### PR TITLE
adding colors management in typography components

### LINKED ISSUE #34: typography & colors
~~currently the Storybook page for Typography components expose a `color` control, and that prop is accepted by the components but its not used in the code.~~

~~This is a leftover of an unfinished discussion regarding how we want to handle colors for typography (accept any color value thus breaking the DS guidelines or expose a subset via props?).~~

~~### Quick fix
remove the color property or set it to accept any color value and set it as `style={}` on the element~~

We have defined with the designers how to handle the colors in typography

### Requirement
All of the typography components should expose a prop that allow a user of the components to choose one of the colors available in the tokens file.

No restrictions will be applied for the moments except that an user can't apply whatever color they want outside of the ones being part of the DS

### ⚠️ warning
some colors are still missing in the design token file

### PR DESCRIPTION
**Pull request by moebiusmania**
## Github issue
closes #34 

## What
adding a `color` property in every typography component that given a color token path value (_ie: `orange.20`_) renders that given DS color as an inline style.

updated the related stories pages with the proper argument controller and a list of values obtained from the design tokens object.

## Note
- also added the lint check in CI for `PR`s and `main` branch

### CODE REVIEW

**Review by alefiori — APPROVED** [review]
Maybe it could be useful (to improve DX), to use a custom literal type for colors (instead of string). Something like this:
```typescript
type Colors = NestedKeyOf<typeof tokens.foundations.colors>;

type NestedKeyOf<ObjectType extends object> = {
  [Key in keyof ObjectType & string]: ObjectType[Key] extends object
    ? `${Key}.${NestedKeyOf<ObjectType[Key]>}`
    : `${Key}`;
}[keyof ObjectType & string];
```

**Comment by moebiusmania** [comment]
> Maybe it could be useful (to improve DX), to use a custom literal type for colors (instead of string). Something like this:
> 
> ```ts
> type Colors = NestedKeyOf<typeof tokens.foundations.colors>;
> 
> type NestedKeyOf<ObjectType extends object> = {
>   [Key in keyof ObjectType & string]: ObjectType[Key] extends object
>     ? `${Key}.${NestedKeyOf<ObjectType[Key]>}`
>     : `${Key}`;
> }[keyof ObjectType & string];
> ```

@alefiori this was a nice addition, thanks

File: src/components/Switch/Switch.tsx [hunk 1]
```diff
@@ -1,30 +1,31 @@
...
+  onChange,
+  isSelected,
+  disabled,
+}) => {
   const optionalProps = {
-    ...(onChange ? {onChange: onChange}: {}),
-    ...(typeof isSelected !== 'undefined' ? {isSelected: isSelected}: {})
-  }
+    ...(onChange ? { onChange: onChange } : {}),
+    ...(typeof isSelected !== "undefined" ? { isSelected: isSelected } : {}),
```
Comment (Serse2) [inline src/components/Switch/Switch.tsx:20]: Maybe i don't understand, but why we need this check instead set as `false` like default?

Comment (moebiusmania) [reply src/components/Switch/Switch.tsx:20]: ask @pietrograndi , I've just fixed the linting issues 😄

**Review by Serse2 — APPROVED** [review]

**Review by alefiori — APPROVED** [review]

**Pull request merged**